core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -105,6 +105,17 @@ package object config {
     .bytesConf(ByteUnit.MiB)
     .createOptional
 
+  private[spark] val DRIVER_MEMORY_OVERHEAD_FACTOR =
+    ConfigBuilder("spark.driver.memoryOverheadFactor")
+      .doc("This sets the Memory Overhead Factor on the driver that will allocate memory to " +
+        "non-JVM memory, which includes off-heap memory allocations, non-JVM tasks, various " +
+        "systems processes, and tmpfs-based local directories.")
+      .version("3.3.0")
+      .doubleConf
+      .checkValue(factor => factor > 0,
```

Comment:
Should we also check whether the factor is configured to be larger than 1.0? Same issue applies below for the spark.executor.memoryOverheadFactor? 
We do check this for spark.memory.fraction and spark.memory.storageFraction.

Reply:
https://github.com/apache/spark/pull/35504#discussion_r808320307

I think that's a fraction of existing memory, where this is an amount of additional memory to add, so there's no reason to set an upper bound on this?

Reply:
Oh, right. Yes, there shouldn't be  an upper bound